### DISCUSSION TITLE
Are there the Maximum number of objects per bucket limit by inode of linux system?

### DISCUSSION THREAD

**Discussion by wuweiminhuo** [minio/minio#11391] [Q&A] [answered]
I want to make sure whether the maximum number of objects per bucket is no-limit?
because i want to put ten million images into one bucket without other directory, but when i saw the file location is also create the directory of bucket and the object in the disk, if there are so many images in same directory , should limit by the inode of linux system?  so that is have limit.
could anybody reply my question?

**Comment by klauspost** [comment, marked as answer]
Yes. And 10M files at the same prefix level will likely be slow because of FS limitations.

Please use community slack for similar questions and followup: https://slack.min.io/

**Reply by duongvox99** [reply]
Can you describe more detail about these limitations?

**Reply by klauspost** [reply]
@duongvox99 Depends on the filesystem you use. A single folder with 10M files at the same level will likely be slow on pretty much any file system, so avoid that.

**Reply by duongvox99** [reply]
@klauspost Thank you so much.
